src/App.tsx

```diff
@@ -1,17 +1,113 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './app.typedefs';
+import todos from './api/todos';
 import users from './api/users';
 
+const preparedTodos: PreparedTodo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || undefined,
+}));
+
 const App: React.FC = () => {
+  const [todosList, setTodos] = useState(preparedTodos);
+  const [todosTitle, setTodosTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const addTodo = (title: string, userId: number) => {
+    const todo: PreparedTodo = {
+      userId,
+      id: todosList.length + 1,
```

Comment:
> - `id` is a maximal `id` in the array + 1

not `list.length`
something like this:
`Math.max(...array.map((todo) => todo.id)) + 1`